addons/web_editor/tests/test_controller.py

```diff
@@ -2,16 +2,65 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 import binascii
-
-from odoo.tools.json import scriptsafe as json_safe
-from odoo.addons.http_routing.models.ir_http import slug
+import json
 
 import odoo.tests
-from odoo.tests.common import HttpCase
+from odoo.tests.common import HttpCase, new_test_user
+from odoo.tools import mute_logger
+from odoo.tools.json import scriptsafe as json_safe
 
+from odoo.addons.http_routing.models.ir_http import slug
 
 @odoo.tests.tagged('-at_install', 'post_install')
 class TestController(HttpCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        portal_user = new_test_user(cls.env, login='portal_user', groups='base.group_portal')
+        cls.portal = portal_user.login
+        admin_user = new_test_user(cls.env, login='admin_user', groups='base.group_user,base.group_system')
+        cls.admin = admin_user.login
+
+        cls.headers = {"Content-Type": "application/json"}
+
+        cls.error = "odoo.exceptions.AccessError"
+
+    def _build_payload(self, params=None):
+        """
+        Helper to properly build jsonrpc payload
+        """
+        return {
+            "jsonrpc": "2.0",
+            "method": "call",
+            "id": 0,
+            "params": params or {},
+        }
+
+    def test_portal_fetch_attachment(self):
+        self.authenticate(self.portal, self.portal)
+        payload = self._build_payload()
+        response = self.url_open('/web_editor/attachment/fetch', data=json.dumps(payload), headers=self.headers)
+        self.assertEqual(200, response.status_code)
+        self.assertNotIn(self.error, response.text)
```

Comment:
Apply the review everywheren not only on the line from which the comment originate :)
https://github.com/odoo/odoo/pull/82612#discussion_r847407264